File: kleddit-node-server/src/user/utils/PasswordHash.ts
import * as bcryptjs from 'bcryptjs'

class PasswordHash {

    public hashWithSalt(password: string): string {
        const salt = bcryptjs.genSaltSync(10)
        return bcryptjs.hashSync(password, salt)
    }

    public matches(password: string, hash: string): boolean {
        return bcryptjs.compareSync(password, hash)
    }

}

export default PasswordHash

File: kleddit-spring-server/src/main/java/com/soze/kleddit/user/password/PasswordHash.java
package com.soze.kleddit.user.password;

import org.mindrot.jbcrypt.BCrypt;
import org.springframework.stereotype.Service;

@Service
public class PasswordHash {

  /**
   * Hashes a given password with a salt.
   *
   * @param password password to hash
   * @return salted hash
   */
  public String hashWithSalt(char[] password) {
    return BCrypt.hashpw(String.copyValueOf(password), BCrypt.gensalt(10));
  }

  public boolean matches(char[] password, String hash) {
    return BCrypt.checkpw(String.copyValueOf(password), hash);
  }

}
